pkg/api/types.go

```diff
@@ -983,6 +994,16 @@ type Binding struct {
 	Target ObjectReference `json:"target"`
 }
 
+// DeleteOptions may be provided when deleting an API object
+type DeleteOptions struct {
+	TypeMeta `json:",inline"`
+
+	// Optional duration in seconds before the object should be deleted. Value must be non-negative integer.
+	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
+	// specified type will be used.
+	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
```

Comment:
Brainstorming other options:
* -1 to imply either immediate deletion or default grace period
* use a string
* use time.Duration

I maybe could get on board with time.Duration (here and elsewhere). It would be similar in spirit to the resource quantity type.